test/serializers/attribute_test.rb

```diff
@@ -33,6 +33,21 @@ def test_multiple_calls_with_the_same_attribute
 
         assert_equal([:title], serializer_class._attributes)
       end
+
+      def test_type_attribute_override
+        attribute_serializer = Class.new(ActiveModel::Serializer) do
+          attribute :id, key: :type
```

Comment:
nitpick: it's not an override anymore if the internal name is now 'json_api_type', right?